lib/smart_answer/calculators/uk_benefits_abroad_calculator.rb

```diff
@@ -106,6 +106,11 @@ def employer_paying_ni_not_ssp_country_entitled?
       (COUNTRIES_OF_FORMER_YUGOSLAVIA + %w[barbados guernsey jersey israel turkey]).include?(country)
     end
 
+    def social_security_countries?
```

Comment:
This seems to be a helper method in place of having to write `social_security_countries_jsa? && social_security_countries_iidb? && ...`.

Seems a good and useful idea, but rather than duplicate the country names I'd be tempted to call these submethods as above, so that there is less we have to keep in sync. E.g. if a country got added to `social_security_countries_jsa?` we'd also have to remember to update `social_security_countries?`, which feels like a potentially dangerous gap we could miss.

Reply:
Just looking into this. I agree this is a dangerous gap. I was looking at how to fix it. 

In the country question for those going abroad we use `social_security_countries_jsa?` to direct `COUNTRIES_OF_FORMER_YUGOSLAVIA guernsey jersey new-zealand` to ask if they have worked or lived int he EEA. 

but also in the country question we use `social_security_countries?` to direct `COUNTRIES_OF_FORMER_YUGOSLAVIA barbados bermuda canada guernsey jersey israel jamaica mauritius new-zealand philippines turkey usa` to `jsa_social_security_already_abroad_outcome`.

These countries are the same as used in the method `social_security_countries_bereavement_benefits?` however `social_security_countries?` is being used in the jsa journey so I don't really want to use the `social_security_countries_bereavement_benefits?` method as it may get confusing later if we just want to change the bereavement_benefits journey.

Not sure what would be the best way to distinguish the 2 methods from each other, would renaming the methods to `social_security_countries_jsa_going_abroad?` and `social_security_countries_jsa_already_abroad?` be an acceptable solution? Or is there a better direction to take this in?

Reply:
Ah, I understood `social_security_countries?` to mean "all countries which have some form of UK social security". In which case you'd just call all the relevant submethods and tack a `.uniq` on the end to filter out any duplicate country names. If it's not actually used like that, then I'd just remove it - better to refer explicitly to the relevant 'submethods' rather than rely on the wrong abstraction.

I've re-read your comment a few times and still don't really understand the nuances, so I suspect we're better off just removing the method if it isn't very helpful! But more generally... it sounds like perhaps you might want to consider a method with an argument:

```ruby
def social_security_countries_jsa(already_abroad: false)
  # return different lists depending on the `already_abroad` argument
end
```

Reply:
Sorry I don't think I've explained it super well! But I really like using`def social_security_countries_jsa(already_abroad: false)` I think that makes it clearer! 

I'll make those changes now! Thank you! :)